Dockerfile

```diff
@@ -1,19 +1,26 @@
 FROM ubuntu:focal-20210827
 
+ARG USER=haskell
+ARG UID=1000
+
 ARG DEBIAN_FRONTEND=noninteractive
 ARG STACK_VERSION=2.7.3
 
 ENV LANG=C.UTF-8
 
-# Make sure this path includes initdb for the usage of tmp-postgres
-ENV PATH=/usr/lib/postgresql/13/bin/:$PATH
+ENV PATH=/home/haskell/.ghcup/bin:/stack/bin:$PATH
+
+# Create a default home for the default user & allow any user to sudo
+RUN useradd --create-home --uid "$UID" "$USER" \
+  && echo "$USER ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
 
-RUN apt-get update -y && apt-get install -y gnupg curl ca-certificates
-RUN curl https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor | tee /etc/apt/trusted.gpg.d/apt.postgresql.org.gpg >/dev/null
-RUN echo "deb http://apt.postgresql.org/pub/repos/apt focal-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+# Have a default work directory. Chances are your configs will override this to provide a better
+# experience like terminal click to go to definition.
+WORKDIR "/home/$USER"
 
 RUN apt-get update -y \
   && apt-get install -y \
+  --option=Dpkg::Options::="--force-confold" \
```

Comment:
What does this do?

Reply:
Good catch, that's more compatible and less likely to break things.